tasks/upload_processor.py

```diff
@@ -372,22 +352,23 @@ def process_impl_within_lock(
                     )
 
             for processed_individual_report in processings_so_far:
-                deleted_archive = self._possibly_delete_archive(
-                    processed_individual_report, report_service, commit
-                )
-                if not deleted_archive:
-                    self._rewrite_raw_report_readable(
-                        processed_individual_report, report_service, commit
+                upload = processed_individual_report.pop("upload_obj", None)
+                raw_report = processed_individual_report.pop("raw_report", None)
```

Comment:
I think this could be moved inside the `if upload` conditional as it's only used in the `_rewrite_raw_report_readable` fn.

Reply:
The `pop` should happen regardless though. We want to always pop both fields off the dict, even if `upload` happens to be non-existant.